front/src/common/Map/WarpedMap/core/helpers.ts

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable prefer-destructuring, no-plusplus */
+import { Feature, FeatureCollection, LineString, Point, Polygon, Position } from 'geojson';
+import _, { clamp, first, last, mapValues } from 'lodash';
+import length from '@turf/length';
+import { point } from '@turf/helpers';
+import along from '@turf/along';
+import distance from '@turf/distance';
+import { MapGeoJSONFeature } from 'maplibre-gl';
+
+import { EditoastType, LAYER_TO_EDITOAST_DICT, LayerType } from 'applications/editor/tools/types';
+import { getMixedEntities } from 'applications/editor/data/api';
+import { flattenEntity } from 'applications/editor/data/utils';
+import vec, { Vec2 } from 'common/Map/WarpedMap/core/vec-lib';
+
+/**
+ * Useful types:
+ */
+export type TriangleProperties = { triangleId: string };
+export type Triangle = Feature<Polygon, TriangleProperties>;
+export type BarycentricCoordinates = [number, number, number];
+export type GridFeature = FeatureCollection<Polygon, TriangleProperties>;
+export type GridIndex = Record<string, Triangle>;
+export type Grids = {
+  regular: GridFeature;
+  warped: GridFeature;
+};
+export type PointsGrid = Record<number, Position>[];
+
+/**
+ * Path manipulation helpers:
+ */
```

Comment:
Be carreful with the comments, with 2* you create documentation :)
```suggestion
/*
 * Path manipulation helpers:
 */

/** Given a line and a number N, cut the line into N samples (so compute N-1 points) and return the step between 2 points and the array of points */
```